src/Columns/ColumnUnique.h

```diff
@@ -361,6 +361,17 @@ size_t ColumnUnique<ColumnType>::uniqueInsert(const Field & x)
 
     FieldVisitorGetData visitor;
     applyVisitor(visitor, x);
+    if (valuesHaveFixedSize())
+    {
+        if constexpr (std::endian::native == std::endian::big)
+        {
+            if ((x.getType() == Field::Types::Int64 || x.getType() == Field::Types::UInt64 || x.getType() == Field::Types::Bool) && size_of_value_if_fixed < 8)
```

Comment:
The PR description says that the problem occurs if "short(non-64 bit) integer Fields are inserted into ColumnUnique". Can it also occur if 32 or 16 bit fields with significant byte(s) = 0 are inserted?